tools/migration/cfg/migrate.py

```diff
@@ -24,9 +25,12 @@ def main():
     input_version =  utils.get_conf_version(args.input_path)
```

Comment:
1) input_version could be 'unsupported', but seems we don't handle this case here.
2) Better to ensure target_version > input_version to do migration.
3) The logic in get_conf_version() is a little cool. Can we check the greatest tag of a harbor container image (e.g. harbor-log) to get current installed version?

Reply:
for 1) and 2) currently it's all handled in `search`, which tries to find a handler chain to migrate the .cfg file from source to target.  So both cases will fail as expected, if you are suggesting refining error messages, those are cosmetic change can be made in future.

for 3) For integration with VIC appliance, this script has to be executed in a container to which the harbor.cfg is single source of truth.

Reply:
Got it. Can we handle "input_version == target_version" in search too ?

Reply:
I think a message is needed to indicate why upgrade is not needed, at least print out the source version and target version.

Reply:
So then let's keep it as is for now.